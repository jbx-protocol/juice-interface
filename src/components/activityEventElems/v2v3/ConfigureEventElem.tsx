import { t, Trans } from '@lingui/macro'
import { MinimalCollapse } from 'components/MinimalCollapse'
import RichNote from 'components/RichNote'
import { BigNumber } from 'ethers'
import {
  V2V3FundingCycle,
  V2V3FundingCycleMetadata,
} from 'models/v2v3/fundingCycle'

import FundingCycleDetails from 'components/v2v3/V2V3Project/V2V3FundingCycleSection/FundingCycleDetails'
import { PV_V2 } from 'constants/pv'
import { V2V3ProjectContext } from 'contexts/v2v3/Project/V2V3ProjectContext'
import { ProjectEventsQuery } from 'generated/graphql'
import useProjectDistributionLimit from 'hooks/v2v3/contractReader/useProjectDistributionLimit'
import { useContext } from 'react'
import { ActivityEvent } from '../ActivityElement'

export default function ConfigureEventElem({
  event,
  withProjectLink,
}: {
  event: ProjectEventsQuery['projectEvents'][0]['configureEvent']
  withProjectLink?: boolean
}) {
  if (!event) return null

  const { primaryETHTerminal } = useContext(V2V3ProjectContext)

  const { data: distributionLimit } = useProjectDistributionLimit({
    projectId: event.projectId,
    terminal: primaryETHTerminal,
    configuration: event.configuration?.toString(),
  })

  const fundingCycle: Partial<V2V3FundingCycle> = {
    duration: BigNumber.from(event.duration),
    weight: BigNumber.from(event.weight),
    discountRate: BigNumber.from(event.discountRate),
    ballot: event.ballot,
    start: event.mustStartAtOrAfter
      ? BigNumber.from(event.mustStartAtOrAfter)
      : undefined,
  }

  const fundingCycleMetadata: Partial<V2V3FundingCycleMetadata> = {
    reservedRate: BigNumber.from(event.reservedRate),
    redemptionRate: BigNumber.from(event.redemptionRate),
    global: {
      pauseTransfers: event.transfersPaused,
      allowSetController: event.setControllerAllowed,
      allowSetTerminals: event.setTerminalsAllowed,
    },
    allowControllerMigration: event.controllerMigrationAllowed,
    allowTerminalMigration: event.terminalMigrationAllowed,
    pausePay: event.pausePay,
    pauseRedeem: event.redeemPaused,
    pauseBurn: event.burnPaused,
    pauseDistributions: event.distributionsPaused,
    useTotalOverflowForRedemptions: event.useTotalOverflowForRedemptions,
    holdFees: event.shouldHoldFees,
    dataSource: event.dataSource,
    useDataSourceForPay: event.useDataSourceForPay,
    useDataSourceForRedeem: event.useDataSourceForRedeem,
    allowMinting: event.mintingAllowed,
  }

  return (
    <ActivityEvent
      event={event}
      header={t`Edited cycle`}
      withProjectLink={withProjectLink}
      pv={PV_V2}
      subject={
        <MinimalCollapse
          header={
            <span className="font-normal">
              <Trans>Details</Trans>
            </span>
          }
        >
          <FundingCycleDetails
            fundingCycle={fundingCycle as V2V3FundingCycle}
            fundingCycleMetadata={
              fundingCycleMetadata as V2V3FundingCycleMetadata
            }
            distributionLimit={distributionLimit?.[0]}
            distributionLimitCurrency={distributionLimit?.[1]}
            mintRateZeroAsUnchanged
          />
          {event.memo ? <RichNote className="mt-3" note={event.memo} /> : null}
        </MinimalCollapse>
      }
    />
  )
}
