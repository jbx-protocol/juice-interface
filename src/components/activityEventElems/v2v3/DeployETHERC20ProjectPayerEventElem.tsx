import { t, Trans } from '@lingui/macro'
import EthereumAddress from 'components/EthereumAddress'
import RichNote from 'components/RichNote'
import { ProjectEventsQuery } from 'generated/graphql'

import { PV_V2 } from 'constants/pv'
import { ActivityEvent } from '../ActivityElement'

export default function DeployETHERC20ProjectPayerEventElem({
  event,
  withProjectLink,
}: {
  event:
    | ProjectEventsQuery['projectEvents'][0]['deployETHERC20ProjectPayerEvent']
  withProjectLink?: boolean
}) {
  if (!event) return null

  return (
    <ActivityEvent
      event={event}
      header={t`Deployed a project payer address`}
      withProjectLink={withProjectLink}
      pv={PV_V2}
      subject={
        <Trans>
          from <EthereumAddress address={event.from} />
        </Trans>
      }
      extra={
        <div>
          <Trans>
            Address: <EthereumAddress address={event.address} />
          </Trans>
          {event.memo && (
            <div className="mt-2">
              <RichNote note={event.memo} />
            </div>
          )}
        </div>
      }
    />
  )
}
