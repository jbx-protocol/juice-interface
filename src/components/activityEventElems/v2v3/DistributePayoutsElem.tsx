import { t } from '@lingui/macro'
import EthereumAddress from 'components/EthereumAddress'
import { JuiceboxAccountLink } from 'components/JuiceboxAccountLink'
import ETHAmount from 'components/currency/ETHAmount'
import V2V3ProjectHandleLink from 'components/v2v3/shared/V2V3ProjectHandleLink'
import { PV_V2 } from 'constants/pv'
import {
  ProjectEventsQuery,
  useSplitDistributionsForDistributePayoutsEventQuery,
} from 'generated/graphql'
import { client } from 'lib/apollo/client'

import { ActivityEvent } from '../ActivityElement'

export default function DistributePayoutsElem({
  event,
  withProjectLink,
}: {
  event: ProjectEventsQuery['projectEvents'][0]['distributePayoutsEvent']
  withProjectLink?: boolean
}) {
  // Load individual DistributeToPayoutSplit events, emitted by internal transactions of the DistributeReservedPayouts transaction
  const { data } = useSplitDistributionsForDistributePayoutsEventQuery({
    client,
    variables: {
      distributePayoutsEvent: event?.id,
    },
  })

  if (!event) return null

  return (
    <ActivityEvent
      event={event}
      header={t`Sent payouts`}
      withProjectLink={withProjectLink}
      pv={PV_V2}
      subject={
        data?.distributeToPayoutSplitEvents.length ? (
          <span className="font-heading text-lg">
            <ETHAmount amount={event.distributedAmount} />
          </span>
        ) : null
      }
      extra={
        <div>
          {data?.distributeToPayoutSplitEvents.map(e => (
            <div
              key={e.id}
              className="flex items-baseline justify-between text-sm"
            >
              <div className="font-medium">
                {e.splitProjectId ? (
                  <V2V3ProjectHandleLink
                    className="text-grey-900 dark:text-slate-100"
                    projectId={e.splitProjectId}
                  />
                ) : (
                  <JuiceboxAccountLink
                    className="text-grey-900 dark:text-slate-100"
                    address={e.beneficiary}
                  />
                )}
                :
              </div>

              <div className="text-secondary">
                <ETHAmount amount={e.amount} />
              </div>
            </div>
          ))}

          {event.beneficiaryDistributionAmount?.gt(0) && (
            <div className="flex items-baseline justify-between">
              <div>
                <EthereumAddress
                  className="text-grey-900 dark:text-slate-100"
                  address={event.beneficiary}
                />
                :
              </div>
              <div className="text-sm text-grey-500 dark:text-grey-300">
                <ETHAmount amount={event.beneficiaryDistributionAmount} />
              </div>
            </div>
          )}
        </div>
      }
    />
  )
}
