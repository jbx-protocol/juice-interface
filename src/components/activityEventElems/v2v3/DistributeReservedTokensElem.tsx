import { Trans } from '@lingui/macro'
import EthereumAddress from 'components/EthereumAddress'
import { JuiceboxAccountLink } from 'components/JuiceboxAccountLink'
import { PV_V2 } from 'constants/pv'
import { V1ProjectContext } from 'contexts/v1/Project/V1ProjectContext'
import {
  ProjectEventsQuery,
  useSplitDistributionsForDistributeReservedTokensEventQuery,
} from 'generated/graphql'
import { client } from 'lib/apollo/client'
import { useContext } from 'react'
import { formatWad, fromWad } from 'utils/format/formatNumber'
import { tokenSymbolText } from 'utils/tokenSymbolText'
import { ActivityEvent } from '../ActivityElement'

export default function DistributeReservedTokensEventElem({
  event,
  withProjectLink,
}: {
  event: ProjectEventsQuery['projectEvents'][0]['distributeReservedTokensEvent']
  withProjectLink?: boolean
}) {
  const { tokenSymbol } = useContext(V1ProjectContext)

  // Load individual DistributeToReservedTokenSplit events, emitted by internal transactions of the DistributeReservedTokens transaction
  const { data } = useSplitDistributionsForDistributeReservedTokensEventQuery({
    client,
    variables: {
      distributeReservedTokensEvent: event?.id,
    },
  })

  if (!event) return null

  return (
    <ActivityEvent
      event={event}
      withProjectLink={withProjectLink}
      pv={PV_V2}
      header={
        <Trans>
          Sent reserved{' '}
          {tokenSymbolText({
            tokenSymbol,
            capitalize: false,
            plural: true,
          })}
        </Trans>
      }
      subject={
        <div className="font-heading text-lg">
          {formatWad(event.tokenCount, { precision: 0 })}{' '}
          {tokenSymbolText({
            tokenSymbol,
            capitalize: false,
            plural: parseInt(fromWad(event.tokenCount) || '0') !== 1,
          })}
        </div>
      }
      extra={
        <div>
          {data?.distributeToReservedTokenSplitEvents.map(e => (
            <div key={e.id} className="flex items-baseline justify-between">
              <div>
                <JuiceboxAccountLink
                  className="text-grey-900 dark:text-slate-100"
                  address={e.beneficiary}
                />
              </div>

              <div className="text-sm text-grey-500 dark:text-grey-300">
                {formatWad(e.tokenCount, { precision: 0 })}{' '}
                {tokenSymbolText({
                  tokenSymbol,
                  capitalize: false,
                  plural: true,
                })}
              </div>
            </div>
          ))}

          {event.beneficiaryTokenCount?.gt(0) && (
            <div className="flex items-baseline justify-between">
              <div>
                <EthereumAddress
                  className="text-grey-900 dark:text-slate-100"
                  address={event.beneficiary}
                />
                :
              </div>
              <div className="text-sm text-grey-500 dark:text-grey-300">
                {formatWad(event.beneficiaryTokenCount, {
                  precision: 0,
                })}
              </div>
            </div>
          )}
        </div>
      }
    />
  )
}
