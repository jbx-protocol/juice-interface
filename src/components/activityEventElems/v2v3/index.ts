export * from './ConfigureEventElem'
export * from './DeployETHERC20ProjectPayerEventElem'
export * from './DistributePayoutsElem'
export * from './DistributeReservedTokensElem'
export * from './SetFundAccessConstraintsEventElem'
