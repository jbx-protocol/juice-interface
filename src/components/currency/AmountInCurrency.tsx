import { CurrencyName } from 'constants/currency'
import { BigNumber } from 'ethers'
import ETHAmount from './ETHAmount'
import USDAmount from './USDAmount'

/**
 *
 * @returns amount in ETH by default. Specified currency otherwise
 */
export function AmountInCurrency({
  currency,
  amount,
  ...props
}: {
  amount: BigNumber | undefined
  currency?: CurrencyName
}) {
  if (currency === 'USD') return <USDAmount amount={amount} {...props} />
  return <ETHAmount amount={amount} {...props} />
}
