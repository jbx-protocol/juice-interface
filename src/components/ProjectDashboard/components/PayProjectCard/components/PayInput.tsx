import { ChevronDownIcon } from '@heroicons/react/24/outline'
import { V2V3CurrencyOption } from 'models/v2v3/currencyOption'
import { FocusEventHandler } from 'react'
import { twMerge } from 'tailwind-merge'
import { V2V3CurrencyName, V2V3_CURRENCY_ETH } from 'utils/v2v3/currency'
import { CurrencyIcon } from '../../ui/CurrencyIcon'
import { usePayInput } from '../hooks/usePayInput'

export type PayInputValue = {
  amount: string
  currency: V2V3CurrencyOption
}

type Props = {
  className?: string
  placeholder?: string
  value?: PayInputValue
  onChange?: (value: PayInputValue) => void
  onBlur?: FocusEventHandler<HTMLInputElement>
  name?: string
}

export const PayInput = ({
  className,
  placeholder,
  value,
  name,
  onChange,
  onBlur,
}: Props) => {
  const {
    value: { amount, currency },
    onInputChange,
    onCurrencyChange,
  } = usePayInput(value, onChange)

  return (
    <div
      className={twMerge(
        'dark:slate-900 flex items-center gap-2 rounded-lg border border-grey-300 bg-white p-2 pl-3 dark:border-slate-500 dark:bg-slate-950',
        className,
      )}
    >
      <CurrencyIcon className="flex-0 h-7 w-7" currency={currency} />
      <input
        data-testid="pay-input-input"
        className="min-w-0 flex-1 bg-transparent text-base outline-none"
        placeholder={placeholder}
        value={amount}
        onChange={onInputChange}
        onBlur={onBlur}
        name={name}
      />
      <div
        role="button"
        data-testid="pay-input-currency-button"
        className={twMerge(
          'flex select-none items-center gap-0.5 rounded py-1.5 pl-2 pr-3 text-xs font-medium transition-colors',
          currency === V2V3_CURRENCY_ETH
            ? 'bg-bluebs-50 text-bluebs-600 dark:bg-bluebs-500 dark:text-bluebs-950'
            : 'bg-melon-50 text-melon-600 dark:bg-melon-500 dark:text-melon-950',
        )}
        onClick={onCurrencyChange}
      >
        <ChevronDownIcon className="h-5 w-5" />
        {V2V3CurrencyName(currency)}
      </div>
    </div>
  )
}
