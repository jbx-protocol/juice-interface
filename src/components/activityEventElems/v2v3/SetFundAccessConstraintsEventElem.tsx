import { t } from '@lingui/macro'
import { AmountInCurrency } from 'components/currency/AmountInCurrency'
import { ProjectEventsQuery } from 'generated/graphql'
import { V2V3CurrencyOption } from 'models/v2v3/currencyOption'
import { V2V3CurrencyName } from 'utils/v2v3/currency'
import { MAX_DISTRIBUTION_LIMIT } from 'utils/v2v3/math'

import { PV_V2 } from 'constants/pv'
import { ActivityEvent } from '../ActivityElement'

export default function SetFundAccessConstraintsEventElem({
  event,
  withProjectLink,
}: {
  event: ProjectEventsQuery['projectEvents'][0]['setFundAccessConstraintsEvent']
  withProjectLink?: boolean
}) {
  if (!event) return null

  return (
    <ActivityEvent
      event={event}
      withProjectLink={withProjectLink}
      pv={PV_V2}
      header={t`Edited payout`}
      subject={
        <div className="font-heading text-lg">
          {event.distributionLimit.eq(MAX_DISTRIBUTION_LIMIT) ? (
            t`Unlimited`
          ) : (
            <AmountInCurrency
              amount={event.distributionLimit}
              currency={V2V3CurrencyName(
                event.distributionLimitCurrency as V2V3CurrencyOption,
              )}
            />
          )}
        </div>
      }
    />
  )
}
