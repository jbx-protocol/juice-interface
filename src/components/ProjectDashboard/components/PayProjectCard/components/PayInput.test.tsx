/**
 * @jest-environment jsdom
 */
import { fireEvent, render, screen } from '@testing-library/react'
import { V2V3_CURRENCY_ETH } from 'utils/v2v3/currency'
import { PayInput } from './PayInput'

describe('PayInput', () => {
  it('renders', () => {
    const { container } = render(<PayInput />)
    expect(container).toMatchSnapshot()
  })

  it('renders with placeholder', () => {
    const { container } = render(<PayInput placeholder="placeholder" />)
    expect(container).toMatchSnapshot()
  })

  it('renders with value', () => {
    const { container } = render(
      <PayInput value={{ amount: '1', currency: V2V3_CURRENCY_ETH }} />,
    )
    expect(container).toMatchSnapshot()
  })

  test('clicking currency button changes currency', () => {
    render(<PayInput />)
    const button = screen.getByTestId('pay-input-currency-button')
    const icon = screen.getByTestId('pay-input-currency-icon')
    expect(button).toHaveTextContent('ETH')
    expect(icon).toHaveClass('bg-bluebs-50 text-bluebs-500')
    fireEvent.click(button)
    expect(button).toHaveTextContent('USD')
    expect(icon).toHaveClass('bg-melon-50 text-melon-600')
    fireEvent.click(button)
    expect(button).toHaveTextContent('ETH')
    expect(icon).toHaveClass('bg-bluebs-50 text-bluebs-500')
  })
})
