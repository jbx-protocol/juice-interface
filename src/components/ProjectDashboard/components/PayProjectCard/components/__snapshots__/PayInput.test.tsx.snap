// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayInput renders 1`] = `
<div>
  <div
    class="dark:slate-900 flex items-center gap-2 rounded-lg border border-grey-300 bg-white p-2 pl-3 dark:border-slate-500 dark:bg-slate-950"
  >
    <div
      class="flex items-center justify-center rounded-full p-1 transition-colors bg-bluebs-50 text-bluebs-500 dark:bg-bluebs-500 dark:text-bluebs-950 flex-0 h-7 w-7"
      data-testid="pay-input-currency-icon"
    >
      <svg
        class="h-full w-full"
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9828 11.8686C19 11.878 19 11.8968 19 11.9155L19.0086 11.8968V11.9249V11.9437C19 11.9531 19 11.9718 19 11.9718C18.9914 11.9812 18.9914 11.9906 18.9914 11.9906C18.9828 12 18.9656 12.0188 18.9571 12.0188H18.9399L12.0687 15.773C12.0515 15.7918 12.0344 15.7918 12.0086 15.7918C11.9828 15.7918 11.9485 15.773 11.9485 15.773L5.0773 12.0188H5.06012C5.04294 12.0094 5.03436 12 5.02577 11.9906H5.01718V11.9812H5.00859C5 11.9718 5 11.9531 5 11.9531V11.9343V11.9061H5.00859C5.00859 11.8924 5.01319 11.8837 5.01903 11.8726C5.02116 11.8686 5.02347 11.8642 5.02577 11.8592V11.8498L11.8883 1.04693L11.9055 1.02816L11.9055 1.02815C11.9141 1.01877 11.9313 1 11.9399 1H11.9571H12C12.0086 1 12.0429 1.00939 12.0429 1.00939H12.0515C12.0687 1.01877 12.0773 1.02816 12.0859 1.03754L12.0859 1.03754L12.1031 1.05631L18.9828 11.8592V11.8686ZM12.0945 17.9505L18.3301 14.3464H18.3129C18.3816 14.3089 18.4589 14.3276 18.5018 14.3933C18.5534 14.459 18.5448 14.5529 18.5018 14.6092L12.1031 22.9437C12.0773 22.9812 12.0344 23 11.9914 23C11.9485 23 11.8798 22.9437 11.8798 22.9437L5.48098 14.6092C5.43804 14.5435 5.43804 14.459 5.48098 14.3933C5.53252 14.3276 5.60982 14.3089 5.66994 14.3464L11.9055 17.9505C11.9656 17.9881 12.0344 17.9881 12.0945 17.9505Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <input
      class="min-w-0 flex-1 bg-transparent text-base outline-none"
      data-testid="pay-input-input"
      value=""
    />
    <div
      class="flex select-none items-center gap-0.5 rounded py-1.5 pl-2 pr-3 text-xs font-medium transition-colors bg-bluebs-50 text-bluebs-600 dark:bg-bluebs-500 dark:text-bluebs-950"
      data-testid="pay-input-currency-button"
      role="button"
    >
      <svg
        aria-hidden="true"
        class="h-5 w-5"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5 8.25l-7.5 7.5-7.5-7.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      ETH
    </div>
  </div>
</div>
`;

exports[`PayInput renders with placeholder 1`] = `
<div>
  <div
    class="dark:slate-900 flex items-center gap-2 rounded-lg border border-grey-300 bg-white p-2 pl-3 dark:border-slate-500 dark:bg-slate-950"
  >
    <div
      class="flex items-center justify-center rounded-full p-1 transition-colors bg-bluebs-50 text-bluebs-500 dark:bg-bluebs-500 dark:text-bluebs-950 flex-0 h-7 w-7"
      data-testid="pay-input-currency-icon"
    >
      <svg
        class="h-full w-full"
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9828 11.8686C19 11.878 19 11.8968 19 11.9155L19.0086 11.8968V11.9249V11.9437C19 11.9531 19 11.9718 19 11.9718C18.9914 11.9812 18.9914 11.9906 18.9914 11.9906C18.9828 12 18.9656 12.0188 18.9571 12.0188H18.9399L12.0687 15.773C12.0515 15.7918 12.0344 15.7918 12.0086 15.7918C11.9828 15.7918 11.9485 15.773 11.9485 15.773L5.0773 12.0188H5.06012C5.04294 12.0094 5.03436 12 5.02577 11.9906H5.01718V11.9812H5.00859C5 11.9718 5 11.9531 5 11.9531V11.9343V11.9061H5.00859C5.00859 11.8924 5.01319 11.8837 5.01903 11.8726C5.02116 11.8686 5.02347 11.8642 5.02577 11.8592V11.8498L11.8883 1.04693L11.9055 1.02816L11.9055 1.02815C11.9141 1.01877 11.9313 1 11.9399 1H11.9571H12C12.0086 1 12.0429 1.00939 12.0429 1.00939H12.0515C12.0687 1.01877 12.0773 1.02816 12.0859 1.03754L12.0859 1.03754L12.1031 1.05631L18.9828 11.8592V11.8686ZM12.0945 17.9505L18.3301 14.3464H18.3129C18.3816 14.3089 18.4589 14.3276 18.5018 14.3933C18.5534 14.459 18.5448 14.5529 18.5018 14.6092L12.1031 22.9437C12.0773 22.9812 12.0344 23 11.9914 23C11.9485 23 11.8798 22.9437 11.8798 22.9437L5.48098 14.6092C5.43804 14.5435 5.43804 14.459 5.48098 14.3933C5.53252 14.3276 5.60982 14.3089 5.66994 14.3464L11.9055 17.9505C11.9656 17.9881 12.0344 17.9881 12.0945 17.9505Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <input
      class="min-w-0 flex-1 bg-transparent text-base outline-none"
      data-testid="pay-input-input"
      placeholder="placeholder"
      value=""
    />
    <div
      class="flex select-none items-center gap-0.5 rounded py-1.5 pl-2 pr-3 text-xs font-medium transition-colors bg-bluebs-50 text-bluebs-600 dark:bg-bluebs-500 dark:text-bluebs-950"
      data-testid="pay-input-currency-button"
      role="button"
    >
      <svg
        aria-hidden="true"
        class="h-5 w-5"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5 8.25l-7.5 7.5-7.5-7.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      ETH
    </div>
  </div>
</div>
`;

exports[`PayInput renders with value 1`] = `
<div>
  <div
    class="dark:slate-900 flex items-center gap-2 rounded-lg border border-grey-300 bg-white p-2 pl-3 dark:border-slate-500 dark:bg-slate-950"
  >
    <div
      class="flex items-center justify-center rounded-full p-1 transition-colors bg-bluebs-50 text-bluebs-500 dark:bg-bluebs-500 dark:text-bluebs-950 flex-0 h-7 w-7"
      data-testid="pay-input-currency-icon"
    >
      <svg
        class="h-full w-full"
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.9828 11.8686C19 11.878 19 11.8968 19 11.9155L19.0086 11.8968V11.9249V11.9437C19 11.9531 19 11.9718 19 11.9718C18.9914 11.9812 18.9914 11.9906 18.9914 11.9906C18.9828 12 18.9656 12.0188 18.9571 12.0188H18.9399L12.0687 15.773C12.0515 15.7918 12.0344 15.7918 12.0086 15.7918C11.9828 15.7918 11.9485 15.773 11.9485 15.773L5.0773 12.0188H5.06012C5.04294 12.0094 5.03436 12 5.02577 11.9906H5.01718V11.9812H5.00859C5 11.9718 5 11.9531 5 11.9531V11.9343V11.9061H5.00859C5.00859 11.8924 5.01319 11.8837 5.01903 11.8726C5.02116 11.8686 5.02347 11.8642 5.02577 11.8592V11.8498L11.8883 1.04693L11.9055 1.02816L11.9055 1.02815C11.9141 1.01877 11.9313 1 11.9399 1H11.9571H12C12.0086 1 12.0429 1.00939 12.0429 1.00939H12.0515C12.0687 1.01877 12.0773 1.02816 12.0859 1.03754L12.0859 1.03754L12.1031 1.05631L18.9828 11.8592V11.8686ZM12.0945 17.9505L18.3301 14.3464H18.3129C18.3816 14.3089 18.4589 14.3276 18.5018 14.3933C18.5534 14.459 18.5448 14.5529 18.5018 14.6092L12.1031 22.9437C12.0773 22.9812 12.0344 23 11.9914 23C11.9485 23 11.8798 22.9437 11.8798 22.9437L5.48098 14.6092C5.43804 14.5435 5.43804 14.459 5.48098 14.3933C5.53252 14.3276 5.60982 14.3089 5.66994 14.3464L11.9055 17.9505C11.9656 17.9881 12.0344 17.9881 12.0945 17.9505Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <input
      class="min-w-0 flex-1 bg-transparent text-base outline-none"
      data-testid="pay-input-input"
      value="1"
    />
    <div
      class="flex select-none items-center gap-0.5 rounded py-1.5 pl-2 pr-3 text-xs font-medium transition-colors bg-bluebs-50 text-bluebs-600 dark:bg-bluebs-500 dark:text-bluebs-950"
      data-testid="pay-input-currency-button"
      role="button"
    >
      <svg
        aria-hidden="true"
        class="h-5 w-5"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5 8.25l-7.5 7.5-7.5-7.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      ETH
    </div>
  </div>
</div>
`;
